src/Validator.php

```diff
@@ -10,110 +10,177 @@
 
 class Validator
 {
+    const DOCUMENT_LENGTH_WITH_CODE = 9;
+    const DOCUMENT_LENGTH_WITHOUT_CODE = 8;
+
     /**
-     * @param $documentId
+     * @param Generator $generator
+     */
+    public function __construct($generator)
+    {
+        $this->generator = $generator;
+    }
+
+    /**
+     * @param string $documentId
      * @return bool
      */
     public function isDNIFormat($documentId)
```

Comment:
why do we want all this format checkers methods to be public? To test them? 
Maybe we could get rid of them or make them private and test only the validator ones (isValidDNI, isValidNIE...)